## Import libraries

In [105]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Display data

In [106]:
startup= pd.read_excel('/content/startup-expansion.xlsx') # POUR XSLX
#startup=startup= pd.read_csv('spreadsheets.csv') POUR CVS
startup

Store ID             City       State Sales Region New Expansion  \
0           1           Peoria     Arizona     Region 2           Old   
1           2          Midland       Texas     Region 2           Old   
2           3          Spokane  Washington     Region 2           Old   
3           4           Denton       Texas     Region 2           Old   
4           5    Overland Park      Kansas     Region 2           Old   
..        ...              ...         ...          ...           ...   
145       146         Paterson  New Jersey     Region 1           New   
146       147      Brownsville       Texas     Region 2           New   
147       148         Rockford    Illinois     Region 1           New   
148       149  College Station       Texas     Region 2           New   
149       150    Thousand Oaks  California     Region 2           New   

     Marketing Spend  Revenue  
0               2601    48610  
1               2727    45689  
2               2768    49554  
3               2759    38284  
4               2869    59887  
..               ...      ...  
145             2251    34603  
146             3675    63148  
147             2648    43377  
148             2994    22457  
149             2431    40141  

[150 rows x 7 columns]

In [107]:
startup.head()

Store ID           City       State Sales Region New Expansion  \
0         1         Peoria     Arizona     Region 2           Old   
1         2        Midland       Texas     Region 2           Old   
2         3        Spokane  Washington     Region 2           Old   
3         4         Denton       Texas     Region 2           Old   
4         5  Overland Park      Kansas     Region 2           Old   

   Marketing Spend  Revenue  
0             2601    48610  
1             2727    45689  
2             2768    49554  
3             2759    38284  
4             2869    59887

## Data Info

In [108]:
startup.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Store ID         150 non-null    int64 
 1   City             150 non-null    object
 2   State            150 non-null    object
 3   Sales Region     150 non-null    object
 4   New Expansion    150 non-null    object
 5   Marketing Spend  150 non-null    int64 
 6   Revenue          150 non-null    int64 
dtypes: int64(3), object(4)
memory usage: 8.3+ KB


In [109]:
startup[['Marketing Spend','Revenue']].describe().round(2)

Marketing Spend   Revenue
count           150.00    150.00
mean           2893.15  39301.43
std             367.86  15465.75
min            1811.00  15562.00
25%            2662.25  21113.50
50%            2898.00  42993.00
75%            3111.50  51145.50
max            3984.00  68828.00

In [110]:
startup['Sales Region'].value_counts()

Sales Region
Region 2    86
Region 1    64
Name: count, dtype: int64

In [111]:
startup['State'].unique()

array(['Arizona', 'Texas', 'Washington', 'Kansas', 'New York', 'Alabama',
       'California', 'Massachusetts', 'New Mexico', 'Mississippi',
       'Oregon', 'Florida', 'Oklahoma', 'New Jersey', 'Utah', 'Colorado',
       'Michigan', 'South Carolina', 'Virginia', 'Ohio', 'New Hampshire',
       'Connecticut', 'Iowa', 'Arkansas', 'Tennessee', 'North Carolina',
       'Georgia', 'Illinois', 'Montana', 'Indiana', 'South Dakota',
       'Louisiana', 'Minnesota', 'Wisconsin', 'Rhode Island'],
      dtype=object)

In [112]:
startup['City'].value_counts()

City
Rochester        2
Killeen          1
Wichita Falls    1
Naperville       1
Clovis           1
                ..
Akron            1
Fullerton        1
Manchester       1
Everett          1
Thousand Oaks    1
Name: count, Length: 149, dtype: int64

In [113]:
startup['State'].value_counts()

State
California        40
Texas             17
Florida           12
Washington         7
Colorado           5
Illinois           5
Georgia            4
Alabama            4
Connecticut        4
New Jersey         4
Arizona            3
Tennessee          3
Iowa               3
Michigan           3
South Carolina     3
Utah               3
Massachusetts      3
Kansas             3
New York           3
Louisiana          2
North Carolina     2
Ohio               2
Virginia           2
Oregon             2
Mississippi        1
New Mexico         1
Arkansas           1
New Hampshire      1
Oklahoma           1
Montana            1
Indiana            1
South Dakota       1
Minnesota          1
Wisconsin          1
Rhode Island       1
Name: count, dtype: int64

In [114]:
startup['New Expansion'].value_counts()

New Expansion
Old    140
New     10
Name: count, dtype: int64

In [115]:
startup.isna().sum() #POUR LES VALEURS NULLES

Store ID           0
City               0
State              0
Sales Region       0
New Expansion      0
Marketing Spend    0
Revenue            0
dtype: int64

In [116]:
startup.duplicated().sum()#POUR VERIFIER LA DUPLICATION

0

## Data Analysis

In [117]:
startup.sample(10)#Afficher random 10 lignes

Store ID              City           State Sales Region New Expansion  \
67         68             Davie         Florida     Region 1           Old   
26         27         Elizabeth      New Jersey     Region 1           Old   
149       150     Thousand Oaks      California     Region 2           New   
141       142             Tempe         Arizona     Region 2           New   
47         48          Murrieta      California     Region 2           Old   
138       139         Lafayette       Louisiana     Region 1           Old   
35         36          Pearland           Texas     Region 2           Old   
103       104           Killeen           Texas     Region 2           Old   
82         83  North Charleston  South Carolina     Region 1           Old   
127       128          Columbus         Georgia     Region 1           Old   

     Marketing Spend  Revenue  
67              3177    45418  
26              2557    45017  
149             2431    40141  
141             2911    48954  
47              2344    16372  
138             2603    40255  
35              3127    20057  
103             2962    43773  
82              3115    18754  
127             2877    55203

# question1:est ce que l'ouverture des nouveaux stores est liée a la rentabilié

1/calcul du Max Revenue
la reponce : Non car la rentabilité dans les nouveaux et anciens stores sont les mêmes

2/calcul du la somme Revenue
la reponce : Non car Old sotres=140 et New stores=10 donc la somme de Old > Somme New



OLD et NEW  stores ont un Revenue elevé Dans california donc la stategie de localisation de stores n'est changé

# **Top 10 data where startup is old groupby State maximum Revenue  **

In [118]:
startup[startup['New Expansion']=='Old'].groupby('State').max()['Revenue'].nlargest(10)

State
Arkansas         68828
Michigan         65475
New York         64906
California       64302
Rhode Island     62337
Alabama          60338
Kansas           59887
Massachusetts    59840
Washington       58951
Virginia         57625
Name: Revenue, dtype: int64

# **Top 10 data where startup is new groupby State maximum Revenue **

In [119]:
startup[startup['New Expansion']=='New'].groupby('State').max()['Revenue'].nlargest(10)

State
Texas         63148
Tennessee     55357
California    52366
Arizona       48954
Illinois      48315
New Jersey    34603
Name: Revenue, dtype: int64

# **Top 10 data where startup is old groupby State somme Revenue  **

In [120]:
startup[startup['New Expansion']=='Old'].groupby('State').sum()['Revenue'].nlargest(10)

State
California     1362468
Texas           554964
Florida         479023
Washington      298013
Alabama         221025
New York        160046
Connecticut     158511
Georgia         157656
Colorado        156495
Michigan        147759
Name: Revenue, dtype: int64

# **Top 10 data where startup is new groupby State somme Revenue  **


In [121]:
startup[startup['New Expansion']=='New'].groupby('State').sum()['Revenue'].nlargest(10)

State
California    141883
Illinois       91692
Texas          85605
Tennessee      55357
Arizona        48954
New Jersey     34603
Name: Revenue, dtype: int64

# question2:La corelation entre Prime net et Expences: le taux de rendment

1.Calculer le prix net

In [122]:
startup['Prime_net']=startup['Revenue']-startup['Marketing Spend']

In [123]:
startup

Store ID             City       State Sales Region New Expansion  \
0           1           Peoria     Arizona     Region 2           Old   
1           2          Midland       Texas     Region 2           Old   
2           3          Spokane  Washington     Region 2           Old   
3           4           Denton       Texas     Region 2           Old   
4           5    Overland Park      Kansas     Region 2           Old   
..        ...              ...         ...          ...           ...   
145       146         Paterson  New Jersey     Region 1           New   
146       147      Brownsville       Texas     Region 2           New   
147       148         Rockford    Illinois     Region 1           New   
148       149  College Station       Texas     Region 2           New   
149       150    Thousand Oaks  California     Region 2           New   

     Marketing Spend  Revenue  Prime_net  
0               2601    48610      46009  
1               2727    45689      42962  
2               2768    49554      46786  
3               2759    38284      35525  
4               2869    59887      57018  
..               ...      ...        ...  
145             2251    34603      32352  
146             3675    63148      59473  
147             2648    43377      40729  
148             2994    22457      19463  
149             2431    40141      37710  

[150 rows x 8 columns]

2.Claculer ROI :le taux de rendement
ROI=(Primr_net/cost)*100

In [124]:
(startup['Prime_net']/startup['Marketing Spend'])*100

0      1768.896578
1      1575.430876
2      1690.245665
3      1287.604204
4      1987.382363
          ...     
145    1437.227899
146    1618.312925
147    1538.104230
148     650.066800
149    1551.213492
Length: 150, dtype: float64

In [125]:
startup['Rendement']=round((startup['Prime_net']/startup['Marketing Spend'])*100,2)# round pour avoir 2 chiffres aprés la vergule

In [126]:
startup['Rendement']

0      1768.90
1      1575.43
2      1690.25
3      1287.60
4      1987.38
        ...   
145    1437.23
146    1618.31
147    1538.10
148     650.07
149    1551.21
Name: Rendement, Length: 150, dtype: float64

In [127]:
startup

Store ID             City       State Sales Region New Expansion  \
0           1           Peoria     Arizona     Region 2           Old   
1           2          Midland       Texas     Region 2           Old   
2           3          Spokane  Washington     Region 2           Old   
3           4           Denton       Texas     Region 2           Old   
4           5    Overland Park      Kansas     Region 2           Old   
..        ...              ...         ...          ...           ...   
145       146         Paterson  New Jersey     Region 1           New   
146       147      Brownsville       Texas     Region 2           New   
147       148         Rockford    Illinois     Region 1           New   
148       149  College Station       Texas     Region 2           New   
149       150    Thousand Oaks  California     Region 2           New   

     Marketing Spend  Revenue  Prime_net  Rendement  
0               2601    48610      46009    1768.90  
1               2727    45689      42962    1575.43  
2               2768    49554      46786    1690.25  
3               2759    38284      35525    1287.60  
4               2869    59887      57018    1987.38  
..               ...      ...        ...        ...  
145             2251    34603      32352    1437.23  
146             3675    63148      59473    1618.31  
147             2648    43377      40729    1538.10  
148             2994    22457      19463     650.07  
149             2431    40141      37710    1551.21  

[150 rows x 9 columns]

# le taux en pourcentage %



** A chaque 1 dollar il gagne 17 dollar pour 1er store et 15 dollar pour 2eme store etc...**

In [128]:
startup['Rendement%']=startup['Prime_net']/startup['Marketing Spend']

In [130]:
startup

Store ID             City       State Sales Region New Expansion  \
0           1           Peoria     Arizona     Region 2           Old   
1           2          Midland       Texas     Region 2           Old   
2           3          Spokane  Washington     Region 2           Old   
3           4           Denton       Texas     Region 2           Old   
4           5    Overland Park      Kansas     Region 2           Old   
..        ...              ...         ...          ...           ...   
145       146         Paterson  New Jersey     Region 1           New   
146       147      Brownsville       Texas     Region 2           New   
147       148         Rockford    Illinois     Region 1           New   
148       149  College Station       Texas     Region 2           New   
149       150    Thousand Oaks  California     Region 2           New   

     Marketing Spend  Revenue  Prime_net  Rendement  Rendement%  
0               2601    48610      46009    1768.90   17.688966  
1               2727    45689      42962    1575.43   15.754309  
2               2768    49554      46786    1690.25   16.902457  
3               2759    38284      35525    1287.60   12.876042  
4               2869    59887      57018    1987.38   19.873824  
..               ...      ...        ...        ...         ...  
145             2251    34603      32352    1437.23   14.372279  
146             3675    63148      59473    1618.31   16.183129  
147             2648    43377      40729    1538.10   15.381042  
148             2994    22457      19463     650.07    6.500668  
149             2431    40141      37710    1551.21   15.512135  

[150 rows x 10 columns]

In [ ]:
#eporter la resultat en fichier csv
#startup.to_csv(path)